process/process.go

```diff
@@ -151,29 +193,185 @@ func pause() error {
 		return err
 	}
 
-	Lock.Unlock()
-
 	CurrentStatus = processStatus.PAUSED
 	log.WithFields(log.Fields{
 		"Status": CurrentStatus,
 	}).Info("Process status")
 
 	for TargetStatus == processStatus.PAUSED {
-		time.Sleep(delay)
+		time.Sleep(pauseDelay)
 	}
 
 	if err := bluetooth.OpenDevice(); err != nil {
 		return err
 	}
 
-	if err := lockContainerUpdates(); err != nil {
-		return err
-	}
-
 	CurrentStatus = processStatus.RUNNING
 	log.WithFields(log.Fields{
 		"Status": CurrentStatus,
 	}).Info("Process status")
 
 	return nil
 }
+
+func getOnlineDevices(a application.Application) (map[string]struct{}, error) {
+	onlineDevices, err := a.Board.Scan(a.ResinUUID)
+	if err != nil {
+		return nil, err
+	}
+
+	log.WithFields(log.Fields{
+		"Number of online devices": len(onlineDevices),
+	}).Info("Processing application")
+
+	return onlineDevices, nil
+}
+
+func getProvisionedDevices(a application.Application) ([]device.Device, error) {
+	db, err := storm.Open(config.GetDbPath())
+	defer db.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	var provisionedDevices []device.Device
+	if err := db.Find("ApplicationUUID", a.ResinUUID, &provisionedDevices); err != nil && err.Error() != index.ErrNotFound.Error() {
+		return nil, err
+	}
+
+	return provisionedDevices, nil
+}
+
+func provisionDevice(a application.Application, localUUID string) []error {
+	log.WithFields(log.Fields{
+		"Local UUID": localUUID,
+	}).Info("Provisioning device")
+
+	resinUUID, name, errs := supervisor.DependentDeviceProvision(a.ResinUUID)
+	if errs != nil {
+		return errs
+	}
+
+	db, err := storm.Open(config.GetDbPath())
+	defer db.Close()
+	if err != nil {
+		return []error{err}
+	}
+
+	d := device.New(a.ResinUUID, a.BoardType, name, localUUID, resinUUID)
+	if err := db.Save(&d); err != nil {
+		return []error{err}
+	}
+
+	log.WithFields(log.Fields{
+		"Name":       d.Name,
+		"Local UUID": d.LocalUUID,
+	}).Info("Provisioned device")
+
+	return nil
+}
+
+func updateDevice(d device.Device) error {
+	db, err := storm.Open(config.GetDbPath())
+	defer db.Close()
+	if err != nil {
+		return err
+	}
+
+	return db.Update(&d)
+}
+
+func sendState(d device.Device) []error {
+	online := true
+	if d.Status == deviceStatus.OFFLINE {
+		online = false
+	}
+
+	return supervisor.DependentDeviceInfoUpdateWithOnlineState(d.ResinUUID, (string)(d.Status), d.Commit, online)
+}
+
+func updateFirmware(d device.Device) []error {
+	online, err := d.Board.Online()
+	if err != nil {
+		return []error{err}
+	} else if !online {
+		return nil
+	}
+
+	filepath, err := getFirmware(d)
+	if err != nil {
+		return []error{err}
+	}
+
+	d.Status = deviceStatus.INSTALLING
+	if err := updateDevice(d); err != nil {
+		return []error{err}
+	}
+	if errs := sendState(d); errs != nil {
+		return errs
+	}
+
+	for i := 1; i <= 3; i++ {
```

Comment:
You should export these magic numbers to a const variable, or a default function param.